### PR TITLE
Adding support to models without soft delete

### PR DESCRIPTION
**Pull request by osiro**
Hey @jordanmaguire not sure how to test this...

Might need your hand...

Feature:

Now we can do:

```ruby
car:
  soft_delete: false
  attributes:
    name:
      type: "string"
      validates:
        presence: true
```

This will not add `acts_as_paranoid` to car model and won't add `deleted_at` columns either.

### CODE REVIEW

File: frontier_generators.gemspec [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+  s.homepage = "https://github.com/thefrontiergroup/frontier_generators"
+  s.rubygems_version = "2.4.6"
+  s.summary = "Comprehensive generators for CRUD"
```
Comment (osiro) [inline frontier_generators.gemspec]: Damn... I actually didn't change this... will revert this guy, don't know why these changes are in here....

File: lib/generators/frontier_model/templates/model.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 class <%= model_configuration.as_constant %> < ActiveRecord::Base
 
-  # Soft delete - uses deleted_at field
```
Comment (jordanmaguire) [inline lib/generators/frontier_model/templates/model.rb]: Don't like this comment?

File: lib/model_configuration/model_configuration.rb [hunk 3]
```diff
@@ -62,11 +65,11 @@ def assign_attributes_from_model_configuration(hash)
     @namespaces = hash[@model_name][:namespaces] || []
     @skip_seeds = hash[@model_name][:skip_seeds] || false
     @skip_ui    = hash[@model_name][:skip_ui] || false
+    @soft_delete = hash[@model_name][:soft_delete].nil? ? true : hash[@model_name][:soft_delete]
```
Comment (jordanmaguire) [inline lib/model_configuration/model_configuration.rb]: would prefer doing the same as the above line.

Comment (jordanmaguire) [inline lib/model_configuration/model_configuration.rb]: ok - booleans and junk. Have discussed @osiro will change the above to be the same so all is consistent.

Could do something like

```ruby
@soft_delete = configuration_for(:soft_delete, default: true)
```

File: lib/model_configuration/model_configuration.rb [hunk 4]
```diff
@@ -58,15 +61,19 @@ def primary_attribute
...
       ModelConfiguration::Attribute::Factory.build_attribute_or_association(self, name, properties)
     end
     # TODO: Assert validity of attributes
     @url_builder = ModelConfiguration::UrlBuilder.new(self)
   end
 
-end
\ No newline at end of file
+  def configuration_for(attr, options={ default: false })
+    attr.nil? ? options[:default] : attr
```
Comment (jordanmaguire) [inline lib/model_configuration/model_configuration.rb]: would prefer attribute, cos I think attr highlights in text editors and junk.

File: spec/lib/support/migration_string_builder_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+describe MigrationStringBuilder do
+
+  let(:builder) { MigrationStringBuilder.new(configuration) }
+  let(:configuration)   { ModelConfiguration.new(test_model_path) }
+  let(:test_model_path) { File.join("spec", "support", "test_model.yaml") }
+
+  describe "#to_s" do
+    subject(:output) { builder.to_s }
+
+    it { should eq "CreateTestModel name:string:index created_at:datetime updated_at:datetime" }
```
Comment (jordanmaguire) [inline spec/lib/support/migration_string_builder_spec.rb]: shouldn't this include deleted_at by default?

Comment (jordanmaguire) [inline spec/lib/support/migration_string_builder_spec.rb]: ok I c

File: spec/lib/support/migration_string_builder_spec.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    describe "option: soft_delete" do
+      before { allow(configuration).to receive(:soft_delete) { soft_delete } }
```
Comment (jordanmaguire) [inline spec/lib/support/migration_string_builder_spec.rb]: would prefer `.and_return(soft_delete)` here but just PP I guess

File: spec/lib/support/migration_string_builder_spec.rb (hunk 5 above)
Comment (osiro) [inline spec/lib/support/migration_string_builder_spec.rb]: That's from  test_model.yaml
